src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx

```diff
@@ -79,7 +79,6 @@ class MultipleEnvironmentSelector extends React.PureComponent {
   replaceSelected(newSelection) {
     this.setState({selectedEnvs: new Set(newSelection)});
     this.props.onChange(newSelection);
-    this.props.onUpdate();
```

Comment:
This was causing the GlobalSelectionHeader state to be wiped which was causing the flickering.